eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/ServiceClientInstantiationCheck.java

```diff
@@ -10,24 +10,69 @@
 import com.puppycrawl.tools.checkstyle.checks.naming.AccessModifier;
 import com.puppycrawl.tools.checkstyle.utils.CheckUtil;
 
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
 /**
  * Verify the classes with annotation @ServiceClient should have following rules:
  * <ol>
  *   <li>No public or protected constructors</li>
  *   <li>No public static method named 'builder'</li>
  *   <li>Since these classes are supposed to be immutable, all fields in the service client classes should be final.</li>
  * </ol>
+ *
+ * All methods that has a @ServiceMethod annotation in a class annotated with @ServiceClient should follow below rules:
+ * <ol>
+ *   <li>Follows method naming pattern. Refer to Java Spec:
+ *   https://jogiles.z19.web.core.windows.net/java-spec-draft/JavaSpec.html#sec-common-service-client-patterns</li>
+ *   <li>Methods should not have "Async" added to the method name</li>
+ *   <ol>Return type of async and sync clients should be as per guidelines:
+ *     <li>The return type for async collection should be of type? extends Flux</li>
+ *     <li>The return type for async single value should be of type? extends Mono</li>
+ *     <li>The return type for sync collection should be of type? extends Stream</li>
+ *     <li>The return type for sync single value should be of type? extends Response</li>
```

Comment:
Formatting here is bad - each line should be `<code>? extends Response</code>`. Check it renders appropriately in JavaDoc output.

Reply:
This doesn't seem resolved.